pkg/assembler/backends/inmem/hasSLSA.go

```diff
@@ -305,8 +305,8 @@ func (c *demoClient) convSLSA(in *hasSLSAStruct) (*model.HasSlsa, error) {
 			BuildType:     in.buildType,
 			SlsaPredicate: in.predicates,
 			SlsaVersion:   in.version,
-			StartedOn:     in.start,
-			FinishedOn:    in.finish,
+			StartedOn:     &in.start,
+			FinishedOn:    &in.finish,
```

Comment:
Shouldn't these be set to nil if the start/finish time are 0?

Reply:
storing them as time.Time{}

Reply:
similar to how we handle the version being nil

Reply:
If the times are optional we need to set them to nil on the output (model.HasSlsa) if they were nil coming in (model.SLSAInputSpec)

Reply:
Yeah, these are leaf types, so they should be stored as input.

For version, namespace, etc., we did a canonicalization to empty string because those were intermediate nodes and reference keys, so we couldn't use nil.

Reply:
Thanks for fixing this up @jeffmendoza! I was thinking to match the others. It makes more sense to keep it nil.